crates/biome_grit_parser/src/lexer/mod.rs

```diff
@@ -0,0 +1,1135 @@
+//! An extremely fast, lookup table based, GritQL lexer which yields SyntaxKind tokens used by the biome Grit parser.
+
+#[rustfmt::skip]
+mod tests;
+
+use biome_grit_syntax::{GritSyntaxKind, GritSyntaxKind::*, TextLen, TextRange, TextSize, T};
+use biome_parser::diagnostic::ParseDiagnostic;
+use std::iter::FusedIterator;
+use std::ops::Add;
+use unicode_bom::Bom;
+
+pub struct Token {
+    kind: GritSyntaxKind,
+    range: TextRange,
+}
+
+impl Token {
+    pub fn kind(&self) -> GritSyntaxKind {
+        self.kind
+    }
+
+    pub fn range(&self) -> TextRange {
+        self.range
+    }
+}
+
+/// An extremely fast, lookup table based, lossless JSON lexer
+#[derive(Debug)]
+pub(crate) struct Lexer<'src> {
+    /// Source text
+    source: &'src str,
+
+    /// The start byte position in the source text of the next token.
+    position: usize,
+
+    diagnostics: Vec<ParseDiagnostic>,
+}
+
+impl<'src> Lexer<'src> {
```

Comment:
I'm noticing a few more methods in the JS lexer that can be removed as well, since they match what's inside the generic lexer. Hope you don't mind if I move them in this PR.

Reply:
I'd prefer not. This PR is already big as I said before, and it would be best to **ease** the job of the reviewers instead of making it harder